Moodboard UI

nina-huangg opened the pull request:
This PR includes changes to the code base and the UI. 

UI:
- added an image inspector panel that displays current positions / dimensions
- allowed images to be deleted without major grid change

codebase:
- code clean up for image components, specifically in creating an image map that maps each id to an image URL, position x, position y, dimensions (width), dimensions (height), and visibility. This change allows the code to not rely on separate arrays, but a map that contains all essential information. Other functionalities like undo can be configured easier with this change.

chwawa reviewed (approved):
Nice work, Nina! The inspector panel was a great idea 🤩

By the way, to add the image description in the inspector, I added an "id" section in the Image Map (so that the code can call properties?.get('id') and feed this id to the backend)

nina-huangg commented:
Update (April 7)
This PR contains several updates to the mood board UI. Here are the following changes:
1. configured inspector component so that when user clicks on an image, the position and dimension coordinates will be displayed and changed dynamically in response to dragging and resizing (as shown in the image below)
<img width="1380" alt="Screenshot 2025-04-07 at 4 31 15 PM" src="https://github.com/user-attachments/assets/62ee201e-974c-4065-8986-b51a4a8bd08f" />

2. moved color palette from the mood board to the inspector and changed up routing to adjust for that

3. fixed the issue of the mood board shifting when an image is deleted. 
- initially the mood board is configured using a grid layout, as a result, when an image is deleted, the rest of the images move up a grid, causing the shifting
- the new configuration randomly creates positions for the images based on the size and width of the mood board, mitigating this issue

4. fixed the bug regarding unselectable images when parts of other images are on the top of the image.

To-do
- Undo functionality. There are some complications with undo, specifically in terms of adding / deleting images. Since this is out of scope, this functionality is currently commented out. If time permits, I will come back and implement this functionality.

nina-huangg commented:
@chwawa 
The inspector panel was previously implemented to grow, using `flex-grow`. Currently, I've changed the styling so that the width is fixed. What do you think of this change? I'm open to either option!
